src/data/equiv/basic.lean

```diff
@@ -1742,6 +1742,25 @@ noncomputable def of_injective {α β} (f : α → β) (hf : injective f) : α 
 of_bijective (λ x, ⟨f x, set.mem_range_self x⟩)
   ⟨λ x y hxy, hf $ by injections, λ ⟨_, x, rfl⟩, ⟨x, rfl⟩⟩
 
+section
+
+variables {α' β' : Type*} (e : perm α') {f' : α' → β'}
+  (f : α' ≃ _root_.set.range f') [decidable_pred (∈ _root_.set.range f')]
+
+def perm.via_set_range : perm β' :=
+perm.subtype_congr (perm_congr f e) (equiv.refl _)
```

Comment:
Could use dot notation here
```suggestion
(perm_congr f e).subtype_congr (equiv.refl _)
```

This def needs a docstring to placate the linter too. I do wonder whether this definition is particularly useful vs using `subtype_congr` directly.